### DISCUSSION TITLE
Desktop access from teleport using GCP managed AD

### DISCUSSION THREAD

**Discussion by rajburnwal07** [gravitational/teleport#23387] [Q&A]
Hi community, has anyone worked on desktop access from teleport using managed GCP AD? I have been facing issue with that. Those two command for installing AD DS and AD CS is required if I'm using managed AD over any cloud ? Those commands are also giving error while executing.

I'm using below doc for accessing desktop from teleport with open source teleport Kubernetes cluster.

https://goteleport.com/docs/desktop-access/active-directory/

Thanks in advance.

**Comment by webvictim** [comment]
We haven't tested Teleport Windows Desktop Access with GCP managed AD - I'm not very familiar with how much access/control it allows you over the domain.

If it works similarly to AWS Managed AD (where they give you an `admin` account with a slightly delegated set of permissions) then theoretically you'd be able to make it work by following the manual installation guide: https://goteleport.com/docs/desktop-access/active-directory-manual/

The automatic install scripts will probably only work on self-hosted Active Directory clusters.

**Reply by rajburnwal07** [reply]
thanks @webvictim Let me follow active-directory manual doc. I'll update here.

**Comment by rajburnwal07** [comment]
@webvictim I have followed manual one but its not able to create new AD user and failing with error access denied.

I have performed as of now.

1. Create managed AD over GCP
2. Enabled LDAPS over created managed AD
3. Create administrative VM in same AD network so that I can managed domain joined VM and successfully joined VM and able to access the VM using AD creds.
4. Problematic part where I'm creating 
     https://goteleport.com/docs/desktop-access/active-directory-manual/#step-17-create-a-restrictive-service-account
     
 "Access is denied" is errorring out when I'm running "New-ADUser  -Name $Name  -SamAccountName $SamAccountName  - AccountPassword $SecureStringPassword  -Enabled $true"  
 
 Now Either I'll have to setup AD configuration over GCP from scratch using VM's or need to use PKI infrastructure provided by AWS itself. This PKI Infrastructure was used in teleport official video as well.
 
 https://aws-quickstart.github.io/quickstart-microsoft-pki/
